### PR TITLE
Export PEM File

### PR DESCRIPTION
**Pull request by mattgle**
## Summary

- Export PEM File

## Screenshots

**Android:**

https://user-images.githubusercontent.com/44204622/196766648-b4952e29-df43-49d0-80fd-68592d39e2ff.mp4

**iOS:**

https://user-images.githubusercontent.com/44204622/196975037-35f1e43e-27f4-437f-a226-9a936acc5816.mp4

## Github Issue
https://github.com/Psychedelic/plug-mobile/issues/499

### CODE REVIEW

File: src/redux/slices/keyring.ts [hunk 1]
```diff
@@ -252,6 +252,37 @@ export const validatePem = createAsyncThunk(
...
+      const instance = KeyRing.getInstance();
+      const response = await instance?.getPemFile(walletId);
+      if (response) {
+        await onSuccess(response);
+      } else {
+        onFailure?.();
+      }
+      onFinish?.();
+    } catch (e: any) {
+      return rejectWithValue({ error: e.message });
```
Comment (ignaciosantise) [inline src/redux/slices/keyring.ts:281]: catch should call onFinish too?
also, `rejectWithValue(e.message);`

File: src/screens/tabs/Profile/modals/ExportPem/components/AccountShowcase/index.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    <View style={style}>
+      <Touchable
+        scale={animationScales.small}
+        onPress={onPress}
+        onLongPress={onLongPress}>
+        <View style={[styles.root, selected && styles.selectedRoot]}>
+          <UserIcon icon={icon} />
+          <View style={styles.leftContainer}>
+            <Text style={FontStyles.Normal}>{title}</Text>
+            <Text style={FontStyles.NormalGray}>{shortAddress(subtitle)}</Text>
```
Comment (ignaciosantise) [inline src/screens/tabs/Profile/modals/ExportPem/components/AccountShowcase/index.tsx:40]: we should call `shortAddress` in the parent component and send the text formatted as prop

File: src/screens/tabs/Profile/screens/Settings/index.tsx [hunk 3]
```diff
@@ -154,10 +161,11 @@ function Settings({ navigation }: ScreenProps<Routes.SETTINGS>) {
           </Text>
         </View>
       </ScrollView>
-      <DeleteWallet modalRef={deleteWalletRef} onDelete={handleDeleteWallet} />
+      <ExportPem modalRef={exportPemRef} />
```
Comment (ignaciosantise) [inline src/screens/tabs/Profile/screens/Settings/index.tsx:164]: each modal we add here is a piece of my heart breaking

File: src/redux/slices/keyring.ts (hunk 1 above)
Comment (mattgle) [reply src/redux/slices/keyring.ts:281]: Added!

File: src/screens/tabs/Profile/screens/Settings/index.tsx (hunk 3 above)
Comment (mattgle) [reply src/screens/tabs/Profile/screens/Settings/index.tsx:164]: :KEKW: Totally bro, don't worry, we will find out the best approach soon!

**Review by ignaciosantise — APPROVED** [review]
🚀

**Pull request merged**